src/js/trove/filelib.js

```diff
@@ -27,6 +30,17 @@
       this.name = name;
       this.fd = fs.openSync(name, (append ? "a" : "w"));
     }
+    var checkISD = RUNTIME.getField(SD, "internal").checkISD;
+    var isISD = RUNTIME.getField(SD, "internal").isISD;
+    function makeDirs(baseDir, tree) {
+      if (!fs.existsSync(baseDir)) fs.mkdirSync(baseDir);
+      if (isISD(tree)) {
+        RUNTIME.ffi.toArray(RUNTIME.getField(tree, "keys-list").app()).forEach(function(sub) {
+          var kid = RUNTIME.getColonField(tree, "get-value").full_meth(tree, sub);
+          makeDirs(baseDir + "/" + sub, kid);
```

Comment:
Should we be more judicious about using `path.join` or whatever the blessed `path` library thing is? I know we have other windows issues but that's probably the right thing to do.

Reply:
I can't tell, tbh: https://nodejs.org/docs/latest/api/fs.html#fsmkdirpath-options-callback gives an example of calling `mkdir` on `"/"` even on Windows...